src/index.ts

```diff
@@ -38,57 +42,61 @@ export class Iterable<T> implements LibIterable<T> {
      * @memberof Iterable
      */
     count(): number {
-        return [...this._source].length;
+        if (Array.isArray(this._source)) {
```

Comment:
Since we're already special casing Array here for fast .length access, maybe it might be worthwhile to also add checks for Set and Map and utilize their .size function(s). Other than Array, Set and Map would be the other common built-in source iterables that would be used.

Reply:
On second thoughts, I decided to remove the special casing for Arrays. Calling count() on an iterable is a code smell, and the correct way to optimize would be to not call this in the first place. Almost all usages of count() in ux-renderer are incorrect and should be replaced with .first(). 

[C#'s Enumerable](https://github.com/microsoft/referencesource/blob/master/System.Core/System/Linq/Enumerable.cs#L341) has a similar naive implementation, so we have precedent :)

Reply:
Thanks. Looks good